pkg/common/x509util/serialnumber.go

```diff
@@ -1,23 +1,28 @@
 package x509util
 
 import (
-	"context"
+	"crypto/rand"
+	"fmt"
+	"math"
 	"math/big"
-	"sync/atomic"
 )
 
-type SerialNumber interface {
-	NextNumber(context.Context) (*big.Int, error)
-}
-
-type serialNumber struct {
-	next int64
-}
+var (
+	maxUint64 = new(big.Int).SetUint64(math.MaxUint64)
+	one       = big.NewInt(1)
+)
 
-func NewSerialNumber() SerialNumber {
-	return &serialNumber{}
-}
+// NewSerialNumber creates a random certificate serial number according to CA/Browser forum spec
+// Section 7.1:
+//   "Effective September 30, 2016, CAs SHALL generate non-sequential Certificate serial numbers greater than
+//   zero (0) containing at least 64 bits of output from a CSPRNG"
+func NewSerialNumber() (*big.Int, error) {
+	// Creates random integer in range [0,MaxUint64)
+	s, err := rand.Int(rand.Reader, maxUint64)
```

Comment:
CA/B forum mandates 64 bit minimum, RFC 5280 mandates 160 bit maximum:
```
   Given the uniqueness requirements above, serial numbers can be
   expected to contain long integers.  Certificate users MUST be able to
   handle serialNumber values up to 20 octets.  Conforming CAs MUST NOT
   use serialNumber values longer than 20 octets.
```

Perhaps we should choose something bigger than 64 bits? 128?

Reply:
Sounds good! I'll change it :+1: